### PR TITLE
feat(added lessThan rule): added lessThan and greaterThan rule

### PR DESCRIPTION
**Pull request by hiteshmungpara**
## Proposed changes

added lessThan and greaterThan rule

## Types of changes

What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/adonisjs/validator/blob/master/CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by thetutlage** [comment]
How about adding the counterpart `greaterThan` as well?

**Comment by hiteshmungpara** [comment]
> How about adding the counterpart `greaterThan` as well?

I have added greaterThan rule.

**Comment by thetutlage** [comment]
Seems like the tests are failing https://github.com/adonisjs/validator/runs/4630947409?check_suite_focus=true#step:5:924

**Comment by hiteshmungpara** [comment]
> Seems like the tests are failing https://github.com/adonisjs/validator/runs/4630947409?check_suite_focus=true#step:5:924

fixed

File: test/validations/greater-than.spec.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    {}
+  )
+}
+
+test.group('greaterThan', () => {
+  validate(greaterThan, test, 20, 18, compile(18))
+
+  test('report error when value fails the greaterThan validation', (assert) => {
+    const reporter = new ApiErrorReporter(new MessagesBag({}), false)
+    greaterThan.validate(2, compile(1).compiledOptions, {
```
Comment (thetutlage) [inline test/validations/greater-than.spec.ts:35]: In this case `2 > 1`. So shouldn't the validation pass?

File: src/Validations/miscellaneous/greaterThan.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    let fieldValue
+
+    if (compiledOptions.ref) {
+      const runtimeFieldValue = refs[compiledOptions.ref].value
+      fieldValue = runtimeFieldValue
+    } else if (compiledOptions.fieldValue) {
+      fieldValue = compiledOptions.fieldValue
+    }
+
+    if (value > fieldValue) {
```
Comment (thetutlage) [inline src/Validations/miscellaneous/greaterThan.ts:47]: If the `value` is greater than `fieldValue` (expectedValue is better name here), then the validation should pass.

For example: I say, the user age must be greater than `18`. So in that case, a user with the age of 20 should pass the validation

**Review by thetutlage — CHANGES_REQUESTED** [review]

File: src/Validations/miscellaneous/greaterThan.ts (hunk 2 above)
Comment (hiteshmungpara) [reply src/Validations/miscellaneous/greaterThan.ts:47]: apka point samaj nahi aya

File: test/validations/greater-than.spec.ts (hunk 1 above)
Comment (hiteshmungpara) [reply test/validations/greater-than.spec.ts:35]: ye bhi samaj me nahi aya.

**Pull request closed without merging**